### PR TITLE
Adds better error message for non-supported OSes

### PR DESCRIPTION
**Pull request by joelgallant**
See how #13 could be confusing to people. I wasn't sure if build.rs is the best place for this kind of this, but asked on discord and it seemed to be the recommended approach.

There's no way this library would / could work on macOS or Windows right?

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+fn main() {
+    if ! cfg!(target_os = "linux") || cfg!(target_os = "freebsd") {
```
Comment (loyd) [inline build.rs]: I believe it should be `!(cfg!(...) || cfg!(...))`.

**Comment by loyd** [comment]
Thanks!

> I wasn't sure if build.rs is the best place for this kind of this

See [compile_error](https://doc.rust-lang.org/std/macro.compile_error.html).

> There's no way this library would / could work on macOS or Windows right?

Probably not right now =(.

**Comment by joelgallant** [comment]
Cool, thanks for the tip (`compile_error`) - moved the error into `lib.rs`

**Comment by loyd** [comment]
Thanks!

**Pull request merged**